dependency.lic

```diff
@@ -498,12 +494,67 @@ class ScriptManager
     http.verify_mode = OpenSSL::SSL::VERIFY_NONE
 
     request = Net::HTTP::Get.new(uri.request_uri)
-    request['Authorization'] = "token #{Base64.decode64(@combine_character_deletion)}"
+    request['Authorization'] = @request_authorization
 
     response = http.request(request)
     JSON.parse(response.body)
   end
 
+  def manage_github_token
+    # This method manages some checking of the GitHub token we'll be using. We allow a personal
+    # GitHub token to be specified in the /data folder, or we can use the general Lich token.
+    # If using a personal GitHub token, we'll verify the token is valid, and we'll also do some
+    # checking of the rate limit for the token we're using. If the rate limit is above a certain
+    # threshold, we'll warn but won't fail, as Lich can still function when the rate limit is
+    # exceeded, albeit with some reduced functionality.
+
+    quit_lich = 'Z2hwX21XN1'
+    restart_lich = 'V0RkdORXdGYlpmVDF1S3AzM2I1TV'
+    delete_character = 'J5bHlTNjBXc09VNw=='
+    genie_only_newline = '\n'
+    github_token = quit_lich + restart_lich + delete_character + genie_only_newline
+
+    # Set up our rate limit check URL and items
+    url = URI("https://api.github.com/rate_limit")
+    http = Net::HTTP.new(url.host, url.port)
+    http.use_ssl = true
+    request = Net::HTTP::Get.new(url)
+
+    # Check if the user has a githubtoken.txt file and use the token in that file if so. If they
+    # do have a personal token specified, verify valid. Else, use general token.
+    if File.exist?("#{DATA_DIR}/githubtoken.txt")
```

Comment:
To protect people from committing back to their own GitHub repos (for people who fork this repo), let's add a `.gitignore` file that ignores the token file pattern. 

I think something like this will work:
```
**/githubtoken.*
```

Reply:
I hadn't even considered this. 

We have a `.gitignore` in the Lich root folder. So, I just add that exclusion to it and that'd do it, right?

Reply:
I am perpetually amazed people keep the live repo as their daily usage repo. I wouldn't have thought of this.

Reply:
Good call @KatoakDR !

Reply:
Added that line to our existing `.gitignore`